src/openforms/js/components/admin/form_design/registrations/objectsapi/GenericObjectsApiVariableConfigurationEditor.js

```diff
@@ -0,0 +1,160 @@
+import {FieldArray, useFormikContext} from 'formik';
+import isEqual from 'lodash/isEqual';
+import PropTypes from 'prop-types';
+import React, {useContext} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useAsync, useToggle} from 'react-use';
+
+import {APIContext} from 'components/admin/form_design/Context';
+import {REGISTRATION_OBJECTS_TARGET_PATHS} from 'components/admin/form_design/constants';
+import Field from 'components/admin/forms/Field';
+import FormRow from 'components/admin/forms/FormRow';
+import Select, {LOADING_OPTION} from 'components/admin/forms/Select';
+import ErrorMessage from 'components/errors/ErrorMessage';
+import {post} from 'utils/fetch';
+
+import {asJsonSchema} from './utils';
+
+export const GenericEditor = ({
+  variable,
+  components,
+  namePrefix,
+  isGeometry,
+  index,
+  mappedVariable,
+  objecttype,
+  objectsApiGroup,
+  objecttypeVersion,
+  TargetPathDisplay,
+}) => {
+  const {csrftoken} = useContext(APIContext);
+  const [jsonSchemaVisible, toggleJsonSchemaVisible] = useToggle(false);
+
+  const {
+    loading,
+    value: targetPaths,
+    error,
+  } = useAsync(
+    async () => {
+      const response = await post(REGISTRATION_OBJECTS_TARGET_PATHS, csrftoken, {
+        objectsApiGroup,
+        objecttype,
+        objecttypeVersion,
+        variableJsonSchema: asJsonSchema(variable, components),
+      });
+      if (!response.ok) {
+        throw new Error('Error when loading target paths');
+      }
+
+      return response.data;
+    },
+    // Load only once:
+    []
+  );
+
+  const getTargetPath = pathSegment => targetPaths.find(t => isEqual(t.targetPath, pathSegment));
+
+  const choices =
+    loading || error
+      ? LOADING_OPTION
+      : targetPaths.map(t => [JSON.stringify(t.targetPath), <TargetPathDisplay target={t} />]);
```

Comment:
With the default value of `[]` for the `targetPaths`, we don't need the LOADING_OPTION choices :)
```suggestion
  const choices = targetPaths.map(t => [JSON.stringify(t.targetPath), <TargetPathDisplay target={t} />]);
```

Reply:
Adding to this:

1. If using native selects, it's nice to get the visual feedback in the dropdown that the options are still loading
2. If using react-select, you can forward the `isLoading` prop and then it the variant suggested by Robin is better

Reply:
Using native selects so this remains the same.